metabase/metabase
# [MLv2] Filters — Fix filter picker reuses values after changing a column

Fixes #34960

`FilterPicker` is always passing the `filter` object to widget components if it's available (i.e., if we're editing a filter). This resulted in an unexpected behavior when changing a column for an existing filter. Because the `filter` object isn't updated until the "Update filter" button is closed, it contains `values` that makes sense for the original column.

Fixed by _not_ passing the `filter` object to widgets if the column differs from the one in `filterParts`

### To Verify

1. New > Question > Raw Data > Sample Database > Products
2. Create a `=` or `!=` filter for the Product's Vendor column (pick a couple of values from the dropdown)
3. Click the filter and change the column to Category
4. Ensure you don't see selected Vendor values in the Category values list

### Demo

https://github.com/metabase/metabase/assets/17258145/141a1d0e-2584-4b85-b8d1-b177785007bc
